### DISCUSSION TITLE
Externally editing and annotating sidecar

### DISCUSSION THREAD

**Discussion by micahchoo** [immich-app/immich#14414] [Q&A] [answered]
I'm working on an digital-archive focussed tool that integrates with Immich to provide metadata editing and image annotation for single images by passing the asset id in the url. The goal is to create a tool for metadata handling while remaining non-invasive and future-proof in Immich dev.

### I have tested a few proof-of-concept applications
- Uses Immich's API to fetch image assets 
- Displays them in OpenSeadragon for high-resolution viewing
- Integrates Annotorious for image annotation capabilities
- Communicates with Immich's backend via a proxy server ( i couldn't get it to work with cors) 

### This is what I want to do 
CRUD operations on the sidecar XMP files
   - Adding and editing custom fields from standard metadata schemas (Dublin Core, FOAF, etc.)
   - Visual annotation interface via Annotorious to store annotations in the XMP using the W3C standard

## Key Questions for the Community

   - if I write directly to the XMP sidecar in a custom format. What are the things that could break? Could it get overwritten the next time I use the metadata editor from inside Immich? What's the preferred approach for externally handling XMP sidecar files in Immich?
   - I thought about writing to a separate sidecar but I want to make it so that future Immich development of export and sharing images with metadata won't be affected?(essentially how do I not affect portability) 

PS. I am not a developer, all of this is very new to me

**Comment by bo0tzz** [comment, marked as answer]
> Could it get overwritten the next time I use the metadata editor from inside Immich?

Only the fields that are edited from Immich should be changed, any other fields should stay intact. If the whole file gets overwritten please open an issue about that.

> What's the preferred approach for externally handling XMP sidecar files in Immich?

Writing directly to the sidecar file is fine. You do probably want to trigger the right jobs afterward to make sure Immich picks up the changes.
